data-model/src/main/scala/za/co/absa/enceladus/model/properties/propertyType/package.scala

```diff
@@ -48,17 +48,19 @@ package object propertyType {
 
     isValueConforming(suggestedValue) match {
       case Success(_) =>
-      case Failure(e) => new IllegalArgumentException(s"The suggested value $suggestedValue cannot be used.", e)
+      case Failure(e) => throw PropertyTypeValidationException(s"The suggested value $suggestedValue cannot be used: ${e.getMessage}", e)
```

Comment:
This approach returns to the user. Maybe we could catch it sooner? Return Validation somewhere? We could also keep it after the REST API Audit.

```
Typedefinitionerror: [
  simpletype,
  classza.co.absa.enceladus.model.properties.propertyType.package$EnumPropertyType
];nestedexceptioniscom.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannotconstructinstanceof`za.co.absa.enceladus.model.properties.propertyType.package$EnumPropertyType`,
problem: ThesuggestedvalueoptionDcannotbeused: Value'optionD'isnotoneoftheallowedvalues(optionA,
optionB,
optionC).at[
  Source: (PushbackInputStream);line: 7,
  column: 2
](throughreferencechain: za.co.absa.enceladus.model.properties.PropertyDefinition[
  "propertyType"
])
```

Reply:
I have adjusted the ExceptionHandler to include the message from our `PropertyTypeValidationException` if found as a L2 underlying cause, otherwise, the "generic" L0 message is used. The relevant integTest has been changed to reflect this. Hope you like it 🥇